text/0002-numeric-values.md

```diff
@@ -0,0 +1,535 @@
+3. Since both sides are compile-time known, this can be evaluated at compile time.
+   1. The left hand side is internally a float, this causes the right hand side to be converted to the float `5.0`
+   2. The division is excuted for the result
+4. Assign numeric literal `5.0` to the variable `x`, which requires type `u32`
+   1. The numeric literal, despite being represented by the float, can be coerced to the result integer type.
+   2. `x` is given the `u32` value `1`
+
+Converting between floats and integers at compile time is OK, as long as the final result of the calculate can be coerced into the result type. When a numeric literal with decimal point tries to coerce to an integer, it will be a compile error.
+
+```
+var x: u32 = 5 / 6
+// compile error: compile time numeric `0.8333333333333334` cannot coerce to `u32`
+```
+
+This is better for the developer and easier to implement than looking at the type `u32` and forcing all constants to be of that type, as it still would be a compile error.
+
+If undefined behavior is caused by a constant expression, it is a compile error.
+
+```blight
+var x: u32 = 5 / 0
+// compile error: division by zero
+```
+
+## Widening during arithmetic
+
+When integers of different sizes are used in arithmetic, the compiler will widen the smaller integer to the larger integer.
+
+For example, if a variable of type `u32` named `x` existed, and the following code was run:
+
+```blight
+var x: u32 = ...
+var x: u16 = ...
+x + y
+```
+
+During the addition, `y` is widened to a `u32` to match the type of the larger size int `x`. It can be thought of an implicit `.to(u32)`
+
+```blight
+x + y.to(u32)
+```
+
+Another case where this happens:
+
+```blight
+var y = x.to(u64) + u32.max + 123
+```
+
+To break down exactly how the type resolution happens here:
+
+1. Calculate the type for the first addition:
+1. The types of the operands are calculated
+
+   - `x.to(u64)` is a `u64`
+   - `u32.max` is a `u32`
+
+1. Widening of the smaller type to match the larger type, the expression is rewritten as `x.to(u64) + u32.max.to(u64)`
+1. Repeat this process on the resulting `u64` for the second addition
+1. The types of the operands are calculated
+
+   - `x.to(u64) + u32.max.to(u64)` is a `u64`
+   - `123` is an numeric literal, it gets assigned a type of `u64`
+
+1. The types are the same size, so it is okay.
+1. It is rewritten such like `(x.to(u64) + u32.max.to(u64)) + 123.to(u64)`
+
+## Changes to Identifiers
+
+The following variable names are banned as they refer to built-in types, as they would shadow the implicit global.
+
+- `/[uif][0-9]+/`
+
+It is [not 100% decided](#variable-naming) how this impacts identifiers in general. It should probably be the job of that proposal to decide this.
+
+## Compiling `.to(T)`
+
+As there are only number types, this function will do nothing in most cases. It may need to compile to `Number(x)` or `BigInt(x)` depending on the action taken.
```

Comment:
```suggestion
As only number types have been defined as of this RFC, this function will do nothing in most cases. It may need to compile to `Number(x)` or `BigInt(x)` depending on the action taken.
```